examples/api/lib/material/selection_area/selection_area.2.dart

```diff
@@ -0,0 +1,375 @@
+              // Check if this span contains content after the selection.
+              if (globalNewStart == endOffset && newStart < rawText.length) {
+                afterSelection.add(TextSpan(style: child.style, text: rawText.substring(newStart)));
+              }
+            }
+            count += rawText.length;
+          }
+        } else if (child is WidgetSpan) {
+          if (count < startOffset) {
+            beforeSelection.add(child);
+          } else if (count >= endOffset) {
+            afterSelection.add(child);
+          } else {
+            insideSelection.add(child);
+          }
+          // We do not increment the count here because the
+          // SelectionListener's local range offsets do not
+          // account for un-selectable widgets or widgets
+          // not managed by its selection delegate, like the
+          // ones being inserted in this example.
+        }
+        return true;
+      });
+      dataSourceMap[entry.key] = TextSpan(
+        style: dataSourceMap[entry.key]!.style,
+        children: <InlineSpan>[
+          ...beforeSelection,
+          ...insideSelection,
+          ...afterSelection,
+        ],
+      );
+      rangeIndex += 1;
+    }
+    // Rebuild column contents.
+    for (final MapEntry<_LocalSpanRange, TextSpan> entry in dataSourceMap.entries) {
+      newText.add(
+        Text.rich(
+          entry.value,
+        ),
+      );
+    }
+    // Calculate new local ranges for paragraphs. This is necessary to keep the ranges
+    // in sync after cutting/inserting new content into the textspan.
+    final Map<_LocalSpanRange, TextSpan> newData = <_LocalSpanRange, TextSpan>{};
+    int localCount = 0;
+    for (final MapEntry<_LocalSpanRange, TextSpan> entry in dataSourceMap.entries) {
+      // Since the non-selectable widgetspan is not included in the SelectionListeners
+      // local range calculation, we should ignore them in the plain text calculation.
+      final int spanLength = entry.value.toPlainText(includePlaceholders: false).length;
+      newData[(startOffset: localCount, endOffset: localCount + spanLength)] = entry.value;
+      localCount += spanLength;
+    }
+    dataSourceMap.clear();
+    for (final MapEntry<_LocalSpanRange, TextSpan> entry in newData.entries) {
+      dataSourceMap[entry.key] = entry.value;
+    }
+    newData.clear();
+    _textWidgets.clear();
+    _textWidgets.addAll(newText);
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.blueAccent,
+      body: SelectionArea(
+        key: selectionAreaKey,
+        child: SelectionListener(
+          onSelectionChanged: (SelectionDetails selectionDetails) {
+            if (_menuController.isShown) {
+              ContextMenuController.removeAny();
+            }
+            if (selectionDetails.status != SelectionStatus.uncollapsed
+               || !selectionDetails.selectionFinalized) {
```

Comment:
I believe this is the only reason selectionFinalized is introduce right? Have you thought about exposing the state through selectableResgion instead.

for example you can do `selectionAreaKey.selectableRegion.isSelectionOnGoing` or something like that.

Reply:
I will try this out, am also not the biggest fan of the `SelectionEvent`.

Reply:
I tried this but it did not work out as well. When we end a drag or a long press we do not dispatch any selection events, so with `isSelectionOnGoing`, when we receive the last event for a given drag, it is still set to true. Another thought is to make `isSelectionOnGoing`, a `ValueNotifier<bool>`, and a user can listen to it and so something like below:

```dart
selectionAreaKey.selectableRegion.isSelectionOnGoing.addListener(_handleSelectionOnGoing);
void _handleSelectionOnGoing() {
    if (selectionAreaKey.selectableRegion.isSelectionOnGoing.value) {
        return;
    }
    // Show color me red context menu. This requires the user to save the last `SelectionDetails` they received in
    // SelectionListener.onSelectionChanged.
}
```